packages/react-menu/src/stories/MenuAnchorToTarget.stories.tsx

```diff
@@ -3,23 +3,40 @@ import * as React from 'react';
 import { Menu, MenuList, MenuItem, MenuPopover, MenuProps } from '../index';
 
 import { Button } from '@fluentui/react-button';
+import { PopperRefHandle } from '@fluentui/react-positioning';
 
 export const AnchorToCustomTarget = () => {
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const popperRef = React.useRef<PopperRefHandle>(null);
   const [open, setOpen] = React.useState(false);
-  const [target, setTarget] = React.useState<HTMLElement | null>(null);
   const onOpenChange: MenuProps['onOpenChange'] = (e, data) => {
     setOpen(data.open);
   };
 
+  React.useEffect(() => {
+    if (buttonRef.current) {
+      popperRef.current?.setTarget(buttonRef.current);
+    }
+  }, [buttonRef, popperRef]);
+
+  // A callback ref can be used, but the component the ref is attached to
```

Comment:
not sure about this one, with the limitations on renderOrder for the imperative handle to be assigned, should we just not encourage this at all ?